File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var lengthOfLIS = function(nums) {
    let arr = new Array(nums.length).fill(1);
	
    for (let i = 1; i < nums.length; i++) {
        for (let j = 0; j < i; j++) {
            if (nums[j] < nums[i]) arr[i] = Math.max(arr[i], arr[j] + 1);
        }
    }
  
    return Math.max(...arr);
};

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        arr = [1] * len(nums)
        for i in range(1, len(nums)):
            for j in range(i):
                if nums[j] < nums[i]:
                    arr[i] = max(arr[i], arr[j]+1)
        return max(arr)
                